### PR TITLE
Inventory UIs.

### PR DESCRIPTION
**Pull request by ChrisEddy**

### CODE REVIEW

File: edit.js [hunk 1]
```diff
@@ -46,6 +46,9 @@ import {GuardianMesh} from './land.js';
 import {storageHost} from './constants.js';
 import app from './app-object.js';
 import inventory from './inventory.js';
+import Inventory from './preact-ui/index.js';
+
+console.log(Inventory)
```
Comment (avaer) [inline edit.js:51]: Probably no need to log this.

File: preact-xr-ui/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "preact-xr-ui",
+  "version": "0.0.1",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "build": "preact build --src preact-xr-ui --dest preact-xr-ui/build/"
+  },
+  "author": "The Metaverse",
```
Comment (avaer) [inline preact-xr-ui/package.json:9]: Lol.

Maybe set `private: true`

File: vr-ui.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import * as THREE from './three.module.js';
 import {TextMesh} from './textmesh-standalone.esm.js';
 import easing from './easing.js';
 import * as icons from './icons.js';
-
+import { Inventory as InventoryHTML } from './preact-ui/components/Inventory.js';
```
Comment (avaer) [inline vr-ui.js:6]: Is this the built version?

**Review by avaer — COMMENTED** [review]
This looks great on first class, we just need to make sure we're not totally breaking VR with this.

I'm guessing it also needs an `npm run build` script and Github Action.

**Comment by avaer** [comment]
https://github.com/webaverse/app/issues/72

**Comment by ChrisEddy** [comment]
no more Preact, no more GH actions, doing it vanilla.

File: .gitignore [hunk 4]
```diff
@@ -1 +1,3 @@
 node_modules
+preact-xr-ui/build/
```
Comment (avaer) [inline .gitignore:2]: Not needed anymore?

File: components/helpers.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const functionValueExtractor = async (fn) => {
```
Comment (avaer) [inline components/helpers.js:1]: What is the purpose of this?

File: edit.html [hunk 6]
```diff
@@ -1,192 +1,19 @@
...
 <html>
-
 <head>
   <title>xrpackage | edit</title>
   <link href="index.css" rel=stylesheet>
   <link href="https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700&display=swap" rel="stylesheet">
   <link href="https://fonts.googleapis.com/css2?family=Muli:wght@400;600&display=swap" rel="stylesheet">
   <script src="https://kit.fontawesome.com/0735724151.js" crossorigin="anonymous"></script>
-  <!-- <script src="https://rawcdn.githack.com/ethereum/web3.js/a6ddec59e65116853435f203b25cb9c55824d084/dist/web3.min.js"></script> -->
+  <link href="./components/styles/index.css" rel=stylesheet>
```
Comment (avaer) [inline edit.html:9]: If this is used for components, they should have their own tags for it. Otherwise this doesn't work for 3D.

File: index.css [hunk 7]
```diff
@@ -10,7 +10,7 @@
 body {
   margin: 0;
   /* font-family: 'Open Sans'; */
-  font-family: 'Muli';
+  font-family: 'Muli' !important;
```
Comment (avaer) [inline index.css:13]: Probably shouldn't be `!important`

File: index.css [hunk 8]
```diff
@@ -65,7 +65,7 @@ img:not([src]) {
   background-color: #FFF;
   align-items: center;
   user-select: none;
-  z-index: 1;
+  z-index: 10;
```
Comment (avaer) [inline index.css:68]: Why this magic value?

File: planet.js [hunk 9]
```diff
@@ -777,24 +777,26 @@ planet.connect = async ({online = true, roomName: rn, url = null} = {}) => {
...
-    location.search = '';
-  } else { // connect case
-    const response = await fetch(`${worldsHost}/create`, {
-      method: 'POST'
-    })
-    if (response.ok) {
-      const json = await response.json();
-      // console.log(json);
-      location.search = `?w=${json.id}`;
+document.addEventListener('load',  () => {
```
Comment (avaer) [inline planet.js:780]: Why this deep listener?

File: wallet.html [hunk 10]
```diff
@@ -97,9 +97,11 @@ <h1 id=file-name></h1>
 }
 
 let file = null;
-bindUploadFileButton(document.getElementById('input-file'), f => {
-  file = f;
-});
+document.addEventListener('load', () => {
```
Comment (avaer) [inline wallet.html:100]: Why this listener?

File: components/Opaque.js [hunk 11]
```diff
@@ -0,0 +1,4 @@
+const Opaque = (props) => {
```
Comment (avaer) [inline components/Opaque.js:1]: For optimization reasons, it's not going to be a good idea to spread out little abstract components like this.

We can still do this, but I would collect them in a single file.

File: components/Login.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import { tryLogin } from '../login.js';
+
+window.addEventListener('load', async () => {
```
Comment (avaer) [inline components/Login.js:3]: A component file should not be performing global initialization logic.

**Comment by ChrisEddy** [comment]
dead branch

**Pull request closed without merging**